### ISSUE TITLE
Remote - SSH: Concurrent SSH sessions using the same VSCode version and SSH host/user share environment variables

### ISSUE THREAD

**Issue by Princemachiavelli** [microsoft/vscode-remote-release#5272]
- VSCode Version: 1.58.0-insider (user setup) cc528230987ce3ac61fc4f094ae247d5ff24db3c
- Local OS Version: Windows 10
- Remote OS Version: NixOS 21.05
- Remote Extension/Connection Type: Windows_NT x64 10.0.19043

1. On 2+ machines,  in ~/.ssh/config, configure an SSH host to connect to the same remote host with the same user. Also configure the SSH SetEnv option to set environment variables to the remote host. e.g to set GIT_COMMITTER_NAME and GIT_COMMITTER_EMAIL. The values for the SetEnv variables should be different betwen the machines.
2. Start two sessions using Remote - SSH extension connecting to the same host using the same user. The two sessions need to be using the same VSCode version (commit) but can be on different machines.
3. Use the 'env' command to check the environment on each of the remote sessions. One of the sessions will have the the variables correctly set but the other session (seems to be the session that connected last) will show the variable values of the 'first' connection.

If you *don't* use the exact same VScode version, then the variables are not shared/overwritten by the concurrent session. I susspect this is related to the fact that the remote host's ~/.vscode-server directory seperates out each folder based on the VSCode commit ID. I have had several user's test this and it occurs on stable and insiders and occurs on Windows & Linux.

The use case in which this could occur is somewhat common, you have many students/interns that need to connect to a common remote server and it's simply easier to have them share a single remote user since it avoids stuff like permissions issues and just simplifies administration.

**Comment by roblourens** [comment]
Yeah the ssh environment comes from the remote server, which is shared for all connections per user and spawns all extension host processes.

>  it's simply easier to have them share a single remote user since it avoids stuff like permissions issues and just simplifies administration.

wouldn't recommend that
